### PR TITLE
Adds diagnostic if no return statement found in function

### PR DESCRIPTION
**Pull request by markwpearce**
image

If a function explicitly says it will return something, ensure there is a return statement.

This could be improved by checking all branches.

I know there's a BSLint option that basically checks the same thing, but if you don't have a return statement when you say you're going to return a string or an integer, it will crash.

### CODE REVIEW

File: src/bscPlugin/validation/ScopeValidator.spec.ts [hunk 1]
```diff
@@ -2239,6 +2239,95 @@ describe('ScopeValidator', () => {
         });
     });
 
+    describe('expectedReturnStatement', () => {
```
Comment (TwitchBronBron) [inline src/bscPlugin/validation/ScopeValidator.spec.ts:2242]: We need tests for `sub as <returnType>` as well. This is valid code and works:
```brighterscript
sub getName() as string
  return "bob"
end sub
```

Missing return in `sub as <returnType>`  crashes like function.

image

image

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Hey, I added some comments about a few more edge cases we need to test for.

**Review by TwitchBronBron — APPROVED** [review]
I agree, this belongs in bsc because it has runtime implications when these are mismatched.
